Yoast/yoast-components
# Snippet preview title and description progress measurement overhaul

## Summary

This PR can be summarized in the following changelog entry:

* Make the progress bars for the snippet title and description respond to the content of the input fields.
* Multiple spaces and spaces at the beginning and end of the snippet description are no longer taken into account anymore for the description length progress.
* Newlines in the snippet description are replaced by a space before being displayed in the snippet preview.

## Relevant technical choices:
* Move the measurement of the title and description from `wordpress-seo` to `yoast-components` in order to make the progress bars work without a CMS-specific implementation.
* Remove the test sliders for the progress bars, because the progress bars now respond to the content of the input fields.
* Strip multiple spaces and spaces at the beginning and end of the description.
* I've made a distinction between `mapDataToPreview` and `mapDataToMeasurement`. Most of the mapping is taking place in `mapDataToMeasurement`. In `mapDataToPreview` is decided whether to use the hand-written description or, if that one is empty, the generated description. This distinction exists because the generated description shouldn't be measured because the progress bar should be empty when a generated description is being used.
* Although it is not possible anymore to add newlines to the description and title, the replacement of newlines by spaces is kept in this PR (I've discussed this with @atimmer) because sticking lines together without spaces is never a good idea. 

## Test instructions

This PR can be tested by following these steps:
* Link `stories/add-title-and-description-assessment` to `wordpress-seo`. Test whether the progress bars respond to the content of the input fields.
* Add multiple spaces to the description, both in the middle, at the beginning and the end. Multiple spaces shouldn't be counted for the description length, and spaces from the beginning and end should be stripped as well.
* Play around with changing the title and metadescription, reload the page etc. to make sure everything works as expected with mapping from input field to the preview and from the input field to the progress bars.

Fixes https://github.com/Yoast/wordpress-seo/issues/9777
Fixes https://github.com/Yoast/wordpress-seo/issues/9913
